src/system.js

```diff
@@ -184,35 +271,102 @@ module.exports = AFRAME.registerSystem('physics', {
     }
 
     if (this.trackPerf) {
-      const wrapperEndTime = performance.now();
-      this.cumulativeTimeWrapper += wrapperEndTime;
-      this.cumulativeTimeWrapper -= wrapperStartTime;
-      this.cumulativeTimeWrapper -= engineEndTime;
-      this.cumulativeTimeWrapper += engineStartTime;
+      const afterEndTime = performance.now();
+
+      this.statsTickData.before = engineStartTime - beforeStartTime
+      this.statsTickData.engine = engineEndTime - engineStartTime
+      this.statsTickData.after = afterEndTime - engineEndTime
+      this.statsTickData.total = afterEndTime - beforeStartTime
+
+      this.el.emit("physics-tick-data", this.statsTickData)
+
+      this.tickCounter++;
 
       if (this.tickCounter === 100) {
+
+        this.countBodies[this.data.driver]()
+
         if (this.statsToConsole) {
-          console.log(`Avg. physics tick duration (engine): ${this.cumulativeTimeEngine / 100} msecs`);
-          console.log(`Avg. physics tick duration (wrapper): ${this.cumulativeTimeWrapper / 100} msecs`);
+          console.log("Physics body stats:", this.statsBodyData)
         }
 
-        if (this.statsToEvents) {
-          const engine = this.cumulativeTimeEngine / 100
-          const wrapper = this.cumulativeTimeWrapper / 100
-          const total = engine + wrapper
-          this.timerEventDetails.engine = engine.toFixed(2)
-          this.timerEventDetails.wrapper = wrapper.toFixed(2)
-          this.timerEventDetails.total = total.toFixed(2)
-          this.el.emit("physics-tick-timer", this.timerEventDetails)
+        if (this.statsToEvents  || this.statsToPanel) {
+          this.el.emit("physics-body-data", this.statsBodyData)
         }
-        
         this.tickCounter = 0;
-        this.cumulativeTimeEngine = 0;
-        this.cumulativeTimeWrapper = 0;
       }
     }
   },
 
+  countBodiesAmmo() {
+
+    const statsData = this.statsBodyData
+    statsData.manifolds = this.driver.dispatcher.getNumManifolds();
+    statsData.manifoldContacts = 0;
+    for (let i = 0; i < statsData.manifolds; i++) {
+      const manifold = this.driver.dispatcher.getManifoldByIndexInternal(i);
+      statsData.manifoldContacts += manifold.getNumContacts();
+    }
+    statsData.collisions = this.driver.collisions.size;
+    statsData.collisionKeys = this.driver.collisionKeys.length;
+    statsData.staticBodies = 0
+    statsData.kinematicBodies = 0
+    statsData.dynamicBodies = 0
+    
+    function type(el) {
+      return el.components['ammo-body'].data.type
+    }
+
+    this.driver.els.forEach((el) => {
+
+      switch(type(el)) {
+        case TYPE.STATIC:
+          statsData.staticBodies++ 
+          break;
+
+        case TYPE.DYNAMIC:
+          statsData.dynamicBodies++ 
+          break;
+
+        case TYPE.KINEMATIC:
+          statsData.kinematicBodies++ 
+          break;
+        
+        default:
+          console.error("Unexpected body type:", type(el))
+          break;
+      }
+    })
```

Comment:
I don't want to expose the internal physics engine values for static, dynamic etc. on the event interface - esp. in the Cannon case, where these are encoded as "1" and "2".

But yes, I can use a mapping object, rather than switch statements - I made that update.